arbnode/dataposter/data_poster.go

```diff
@@ -192,22 +192,15 @@ func (p *DataPoster[Meta]) getFeeAndTipCaps(ctx context.Context, lastTipCap *big
 
 	elapsed := time.Since(dataCreatedAt)
 	config := p.config()
-	maxFeeCap := new(big.Int).SetUint64(uint64(config.MaxFeeCapGwei * params.GWei))
-	maxFeeCapDoublings := int64(elapsed / config.MaxFeeCapDoubling)
-	// in tests, this could get way too big
-	if maxFeeCapDoublings > 8 {
-		maxFeeCapDoublings = 8
-	}
-	multiplier := new(big.Int).Exp(big.NewInt(2), big.NewInt(maxFeeCapDoublings), nil)
-	maxFeeCap.Mul(maxFeeCap, multiplier)
+	// MaxFeeCap = (BacklogOfBatches^2 * UrgencyGWei^2 + TargetPriceGWei) * GWei
+	maxFeeCap :=
+		arbmath.FloatToBig(
+			(float64(arbmath.SquareUint(backlogOfBatches))*
+				arbmath.SquareFloat(config.UrgencyGwei) +
+				config.TargetPriceGwei) *
+				params.GWei)
 	if arbmath.BigGreaterThan(newFeeCap, maxFeeCap) {
-		logLevel := log.Info
-		if maxFeeCapDoublings >= 3 {
-			logLevel = log.Error
-		} else if maxFeeCapDoublings >= 1 {
-			logLevel = log.Warn
-		}
-		logLevel(
+		log.Warn(
```

Comment:
I've changed this to a warning instead of an error. I'll make sure this commit is double checked by somebody else but I wanted to merge this without another round of back and forth since I'd been delaying it already :)